common/djangoapps/student/forms.py

```diff
@@ -188,6 +188,19 @@ def __init__(
                                 "required": _("To enroll, you must follow the honor code.")
                             }
                         )
+                elif field_name == 'data_sharing_consent':
+                    if field_value == "required":
+                        self.fields[field_name] = TrueField(
+                            error_messages={
+                                "required": _(
+                                    "You must consent to data sharing to register."
```

Comment:
Having a published policy that we can link to here would be best. If we don't publish a policy, we should be clear about what "data sharing" means here—something like "To register an account, you must agree to share data including x and y with edX." (Do we then give them an opportunity to agree or decline?)